### ISSUE TITLE
byte array in request class with OpenAPI extension works on 0.18.0 and not on 0.19.0+

### ISSUE THREAD

**Issue by Holi60k** [quarkusio/quarkus#3537]
**Describe the bug**
I started to migrate an existing Thorntail application to Quarkus. This project has some REST based endpoints. The classes that are used in the REST endpoints are documented with OpenAPI, and we would like to separate our REST interfaces and implementations so we have an ```IExampleRest``` where the methods are documented with OpenAPI or not but in this case does not matter, and there is an implementation ```ExampleRest```. The request class contains a field of a byte array type, for file uploading or "downloading", and here comes a problem.
With this scenario:
* Quarkus 0.19.0 or 0.20.0
* OpenAPI extension
* IExampleRest - ExampleRest implementation
* Request class with byte array field
the following error logs occour.

On 0.18.0 the `mvn clean install` completes with no error, and the application can be started.

On 0.19.0 the following error log is created on `mvn clean install -X`:
https://pastebin.com/16JG9DxF (log is too long to be readable)

On 0.20.0+ the following error log is created on `mvn clean install -X`:
https://pastebin.com/UTfVjpWR

The "workaround" or the real solution which should be used is to replace the byte[] to InputStream, but I don't know why the build fails in two different ways!

**Expected behavior**
Maybe the build should be finished with no errors.

**Actual behavior**
Build fails with different errors on different versions.
What I see is that there were a PR, (https://github.com/quarkusio/quarkus/pull/2983), where something changed with the indexing, and maybe this causing the problem. I like to use the interface and implementation segregation, we are documenting the interfaces and then implementing the interface to call different service methods, and when I was debugging the build phase I saw that maybe causing the error in the `SmallRyeOpenApiProcessor` class.

**To Reproduce**
1. git clone https://github.com/Holi60k/quarkus-openapi-byte-array 
2. mvn clean install -X 
3. See the log
4. Move to 0.18.0
5. mvn clean install
6. Suprise

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: java version "1.8.0_192"
 - Quarkus version or git rev: 0.18.0, 0.19.0, 0.20.0+

**Comment by gsmet** [comment]
Outch, you're unlucky on this one: I think you hit 2 different bugs in 2 different versions.

@mkouba see the 0.19 log file, it seems we should exclude primitives types (and arrays of primitive types) from indexing here: `BeanArchives`. Or there should be a safeguard somewhere. Maybe it has already been taken care of?

The failure for 0.20 has been fixed in https://github.com/smallrye/smallrye-open-api/pull/166 AFAICS but we will need a SmallRye OpenAPI update.

**Comment by Holi60k** [comment]
The question is that if it happens with other primitives too. At this case right know, using InputStream or MultipartFormData would be better for file uploading, but this older project had byte[] :D Hope it will be fixed soon!

**Comment by mkouba** [comment]
Hm, it seems that SM openapi does not attempt to index primitive types anymore: https://github.com/smallrye/smallrye-open-api/blob/master/implementation/src/main/java/io/smallrye/openapi/runtime/scanner/dataobject/IgnoreResolver.java#L259

@gsmet You're right that we should return `null` instead of throwing an ISE. I'll create a separate issue.

**Comment by gsmet** [comment]
@EricWittmann can you check this one will be fixed too (i.e. when having an array of primitives)?

**Comment by EricWittmann** [comment]
Sure thing.

**Comment by EricWittmann** [comment]
It seems that the fix does **not** address this problem.  The latest MP OpenAPI release gets us beyond the String Index Array Exception, but now there is the StreamClosed error.

I'm digging in.  I think ultimately the jandex index available to **smallrye-open-api** has too much stuff in it (everything?).  I suspect that @MikeEdgar is correct that this is the issue:  https://github.com/quarkusio/quarkus/blob/9aff3d16231ade5b9386d5eedff40d0a1c199b67/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java#L108-L116

I will try a few things - one of which will be to change that method to only use `combinedIndexBuildItem` and ignore `combinedIndexBuildItem`.  Another will be to exclude primitive arrays from that particular class lookup.

All that said, I don't know why there is a **StreamClosed** error when trying to lookup `byte[]` as a class in jandex.  Figuring that part out is probably above my pay grade. :)

**Comment by EricWittmann** [comment]
Excluding primitive arrays from the `@Ignore` resolver did the trick.  There may be additional land mines in there due to the inclusion of all java classes in the jandex index....or there may not.  It might be worth having a conversation about `smallryeOpenApiIndex`.  But the following PR fixes this bug:

https://github.com/smallrye/smallrye-open-api/pull/170

**Comment by mkouba** [comment]
@EricWittmann is it `java.io.IOException: Stream closed`? Where does it come from? It should not come from the `BeanArchiveIndex` on the quarkus master.

**Comment by EricWittmann** [comment]
I didn't capture the stack trace - I assume it was the same as before.  I'll have to go back and do it again.  I'll report back here after I've had a chance to do that.  It's also possible I had a stale Quarkus `master`.  When was the **stream closed** problem fixed?

**Comment by EricWittmann** [comment]
I just tried this again and @mkouba is right - the **stream closed** problem is gone.  I was using an older version of quarkus `master`.  I thought I had pulled the latest but I had not.

So trying everything again **without** the fix in https://github.com/smallrye/smallrye-open-api/pull/170 results in a successful build for the reproducer in this bug. 

That said, I think the fix in the PR is still reasonable.  There's no need to lookup the class if it's a primitive array, since we know it won't be found.
